src/util/worker-callback.js

```diff
@@ -0,0 +1,42 @@
+export const workerCallback = function(options) {
+  const transmuxer = options.transmuxer;
+  const endAction = options.endAction || options.action;
+  const callback = options.callback;
+  const message = Object.assign({}, options, {endAction: null, transmuxer: null, callback: null});
+
+  const listenForEndEvent = (event) => {
+    if (event.data.action !== endAction) {
+      return;
+    }
+    transmuxer.removeEventListener('message', listenForEndEvent);
+
+    // transfer ownership of bytes back to us.
```

Comment:
Since the ownership is transferred here, do we need to transfer the ownership again in the media-segment-request users of `workerCallback`?

Reply:
We do for anywhere else the data is referenced.